apps/src/templates/instructions/InstructionsWithWorkspace.jsx

```diff
@@ -0,0 +1,183 @@
+var connect = require('react-redux').connect;
+var utils = require('../../utils');
+var styleConstants = require('../../styleConstants');
+var CodeWorkspaceContainer = require('../CodeWorkspaceContainer');
+var TopInstructions = require('./TopInstructions');
+var instructions = require('../../redux/instructions');
+
+var HEADER_HEIGHT = styleConstants['workspace-headers-height'];
+var RESIZER_HEIGHT = styleConstants['resize-bar-width'];
+
+/**
+ * A component representing the right side of the screen in our app. In particular
+ * it has instructions in the top pane (unless disabled), a resizer, and then
+ * our code workspace component
+ */
+var InstructionsWithWorkspace = React.createClass({
+  propTypes: {
+    showInstructions: React.PropTypes.bool.isRequired,
+    instructionsCollapsed: React.PropTypes.bool.isRequired,
+    instructionsHeight: React.PropTypes.number.isRequired,
+    instructionsMaxHeight: React.PropTypes.number.isRequired,
+
+    generateCodeWorkspaceHtml: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    // only used so that we can rerender when resized
+    return {
+      width: undefined,
+      height: undefined
```

Comment:
I think these need to be renamed to `windowWidth` and `windowHeight` - those are the only properties you ever include in `setState()` calls here.